src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -586,6 +586,63 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
     );
   };
 
+  createPendingBlock = async (previousBlock: Block) => {
+    const options = this.#options;
+    const minerOpts = options.miner;
+    const nextBlock = this.#readyNextBlock(previousBlock);
+    // If the executables has inProgress transactions, we were already
+    // in the middle of mining a block when the pending block was requested.
+    // The block that is currently being mined is what we want to return as pending,
+    // so we'll reset the executables to be what it was when mining started,
+    // i.e. all inProgress executables are moved to the front of the pending executables.
+    const executables =
+      this.transactions.transactionPool.cloneAndResetExecutables();
+
+    // the number of transactions that should be included in this block depends
+    // on a few factors. in "instamine" mode (meaning blockTime = 0), we usually
+    // only mine 1 transaction per block. However, if the miner is currently
+    // paused (from miner_stop being called), whenever they restart the miner
+    // all executable transactions will be mined, so the pending block should
+    // contain all executables that will fit in a block. if we're not in
+    // instamine mode, the miner will always try to fill the block, regardless
+    // of whether the miner is paused or not
+    const instamine = this.#instamine;
+    const paused = this.#isPaused();
+    const maxTransactions =
+      instamine && !paused ? Capacity.Single : Capacity.FillBlock;
+
+    // we don't want any events from mining to be
+    // caught by `this.vm`, so we need to make a new one.
+    const vm = await this.createVmFromStateTrie(
+      this.trie.copy(false),
+      options.chain.allowUnlimitedContractSize,
+      true
+    );
+    // checkpoint to prevent writing any changes to the database
+    await vm.stateManager.checkpoint();
+    const miner = new Miner(minerOpts, executables, vm, this.#readyNextBlock);
+    const pendingBlockPromise = new Promise<Block>(resolve => {
+      miner.on(
```

Comment:
can you use `const { block } = miner.once("block")` here instead of wrapping this in a Promise?

Reply:
Done! Though I had to rearrange a bit to not await `miner.mine`, is that what you intended?

Reply:
Yeah, probably. And i mean to `await miner.once` in my previous comment. My hope was the change would make it simpler and more readable.